hibike/devices/LineFollower/line_follower.cpp

```diff
@@ -18,44 +18,34 @@ void loop() {
 }
 
 
-uint32_t device_update(uint8_t param, uint32_t value) {
-  return ~((uint32_t) 0);
-}
-
+// You must implement this function.
+// It is called when the device receives a Device Write packet.
+// Updates param to new value passed in data.
+//    param   -   Parameter index
+//    data    -   value to write, in bytes TODO: What endian?
+//    len     -   number of bytes in data
+//
+//   return  -   size of bytes written on success; otherwise return 0
 
-uint32_t device_status(uint8_t param) {
-  return ~((uint32_t) 0);
+uint32_t device_write(uint8_t param, uint8_t* data, size_t len) {
+  return 0;
 }
 
 
-// you must implement this function. It is called with a buffer and a maximum buffer size.
-// The buffer should be filled with appropriate data for a DataUpdate packer, and the number of bytes
-// added to the buffer should be returned. 
+// You must implement this function.
+// It is called when the device receives a Device Data Update packet.
+// Modifies data_update_buf to contain the parameter value.
+//    param           -   Parameter index
+//    data_update_buf -   buffer to return data in
+//    buf_len         -   Maximum length of the buffer? TODO: Clarify
 //
-// You can use the helper function append_buf.
-// append_buf copies the specified amount data into the dst buffer and increments the offset
-uint8_t data_update(uint8_t* data_update_buf, size_t buf_len) {
-  if (buf_len < sizeof(uint16_t) * NUM_PINS) {
-    return 0;
-  }
+//    return          -   sizeof(param) on success; 0 otherwise
 
-  // Read sensor
-  for (int i = 0; i < NUM_PINS; i++) {
-      data[i] = analogRead(pins[i]);  
-  }
-  
-  // Append data to packet buffer
-  uint8_t offset = 0;
-  for (int i = 0; i < NUM_PINS; i++) {
-    append_buf(data_update_buf, &offset, (uint8_t *)&data[i], sizeof(uint16_t));
+uint8_t device_read(uint8_t param, uint8_t* data, size_t len) {
+  if (MAX_PAYLOAD_SIZE - len < sizeof(uint8_t)|| param > 2 || param < 0) {
+    return 0;
   }
-
-  return offset;
-}
-
-// You must implement this function.
-// It is called when the BBB sends a message to the Smart Device tellinng the Smart Device to disable itself.
-// Consult README.md, section 6, to see what exact functionality is expected out of disable.
-void device_disable() {
-
+  float *data_float = (float *)data;
+  data_float[0] = analogRead(pins[param]);
```

Comment:
Does this return a float between 0 and 1023, or 0 and 1?

Try to keep it between 0 and 1.